independent-projects/resteasy-reactive/common/processor/src/main/java/org/jboss/resteasy/reactive/common/processor/EndpointIndexer.java

```diff
@@ -884,17 +894,25 @@ private boolean isRunOnVirtualThread(MethodInfo info, boolean blocking, Blocking
                         + "' uses @RunOnVirtualThread but the target JDK version doesn't support virtual threads. Please configure your build tool to target Java 19 or above");
             }
             if (!blocking) {
-                throw new DeploymentException(
-                        "Method '" + info.name() + "' of class '" + info.declaringClass().name()
-                                + "' is considered a non blocking method. @RunOnVirtualThread can only be used on " +
-                                " methods considered blocking");
+                if (blockingAnnotation != null) {
+                    return false;
+                }
```

Comment:
When can this condition happen?

Reply:
When there is a `@Blocking` annotation on the method

Reply:
Unfortunately yes... this whole thing needs to be rewritten (see my comment in the PR)